### PR TITLE
[FLINK-15044][e2e tests] Clean up TpcdsResultComparator

### PR DESCRIPTION
**Pull request by StephanEwen**
## Purpose / Changelog

This cleans up the code in the `TpcdsResultComparator`. The functionality is not changed.
The code cleanup fixed a bug, as a byproduct.

This uncovered some dirty trailing error message in the expected answer set, which is removed in the second commit.

## Verifying this change

This is a cleanup of a test.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**
  - If yes, how is the feature documented? **not applicable**

### CODE REVIEW

File: flink-end-to-end-tests/flink-tpcds-test/src/main/java/org/apache/flink/table/tpcds/utils/TpcdsResultComparator.java [hunk 1]
```diff
@@ -47,180 +47,188 @@
...
+			int lineNum1, int lineNum2) {
+
+		final boolean matchesPair = compareLine(expected1, actual1) && compareLine(expected2, actual2);
+		final boolean matchesCross = compareLine(expected2, actual1) && compareLine(expected1, actual2);
+
+		if (!(matchesPair || matchesCross)) {
+			System.out.println(String.format("[ERROR] Lines %d/%d do not match in any pairing." +
+					"\n - Expected %d: %s\n - Expected %d: %s\n - Actual %d: %s\n - Actual %d: %s",
+				lineNum1, lineNum2, lineNum1, expected1, lineNum2, expected2,
+				lineNum1, actual1, lineNum2, 2));
```
Comment (KurtYoung) [inline flink-end-to-end-tests/flink-tpcds-test/src/main/java/org/apache/flink/table/tpcds/utils/TpcdsResultComparator.java]: 2 should be `actual2`?

Comment (StephanEwen) [reply flink-end-to-end-tests/flink-tpcds-test/src/main/java/org/apache/flink/table/tpcds/utils/TpcdsResultComparator.java]: yes, thanks for spotting this.

**Comment by KurtYoung** [comment]
BTW I've ran this locally and the results looks good to me.

**Comment by StephanEwen** [comment]
I am getting an error at the moment that the log files / out files are non empty, which is strange, since nothing in the query execution changed.

@KurtYoung Did you see something similar?

**Comment by KurtYoung** [comment]
No, what's in the out files?

**Comment by StephanEwen** [comment]
Hmmm, cannot reproduce it any more.
There seem to have been some errors in the log files, but apparently not a deterministic error.

**Comment by StephanEwen** [comment]
Will merge this...

**Comment by leonardBang** [comment]
@StephanEwen Thanks for your effort very much, LGTM +1.
The README.md file in tpcds-tool directory is still empty，should we write something here?
If not we can delete it in this PR。
 BTW，someone submit a PR in veverica repository https://github.com/ververica/tpc-ds-generators/pull/2/commits/2254a33033b0251861156b1bd56adc9b0579a66d ，should we open the entrance to accept other dsden?

**Comment by StephanEwen** [comment]
@leonardBang yes, we need to put a readme in there and add some license related info to the NOTICE file.

**Pull request closed without merging**